src/ui/UserProfile.cpp

```diff
@@ -0,0 +1,329 @@
+                                    return;
+                            }
+
+                            std::vector<DeviceInfo> deviceInfo;
+                            auto devices         = other_res.device_keys.at(user_id);
+                            auto device_verified = cache::getVerifiedCache(user_id);
+
+                            if (device_verified.has_value()) {
+                                    isUserVerified = device_verified.value().is_user_verified;
+                            }
+
+                            std::optional<crypto::CrossSigningKeys> lmk, lsk, luk, mk, sk, uk;
+
+                            if (!res.master_keys.empty())
+                                    lmk = res.master_keys.at(local_user_id);
+                            if (!res.user_signing_keys.empty())
+                                    luk = res.user_signing_keys.at(local_user_id);
+                            if (!res.self_signing_keys.empty())
+                                    lsk = res.self_signing_keys.at(local_user_id);
+                            if (!other_res.master_keys.empty())
+                                    mk = other_res.master_keys.at(user_id);
+                            if (!other_res.user_signing_keys.empty())
+                                    uk = other_res.user_signing_keys.at(user_id);
+                            if (!other_res.self_signing_keys.empty())
+                                    sk = other_res.self_signing_keys.at(user_id);
+
+                            // First checking if the user is verified
+                            if (luk.has_value() && mk.has_value()) {
+                                    // iterating through the public key of local user_signing keys
+                                    for (auto sign_key : luk.value().keys) {
+                                            // checking if the signatures are empty as "at" could
+                                            // cause exceptions
+                                            auto signs = mk->signatures;
+                                            if (!signs.empty() &&
+                                                signs.find(local_user_id) != signs.end()) {
+                                                    auto sign = signs.at(local_user_id);
+                                                    try {
+                                                            isUserVerified =
+                                                              isUserVerified ||
+                                                              (olm::client()->ed25519_verify_sig(
+                                                                sign_key.second,
+                                                                json(mk.value()),
+                                                                sign.at(sign_key.first)));
+                                                    } catch (std::out_of_range) {
```

Comment:
Don't catch by value, use `catch (std::out_of_range&)` instead (same in line 229)